tests/module/memory_test.c

```diff
@@ -0,0 +1,191 @@
+#include <string.h>
+
+#define DIMENSION 100
+
+// get memory usage from redis-server
+long long _get_memory_usage(RedisModuleCtx *ctx) {
+    long long usage;
+    RedisModuleServerInfoData *info = RedisModule_GetServerInfo(ctx, "memory");
+    RedisModuleString *memory = RedisModule_ServerInfoGetField(ctx, info, "used_memory");
+    RedisModule_StringToLongLong(memory, &usage);
+    RedisModule_FreeServerInfo(ctx, info);
+    RedisModule_FreeString(ctx, memory);
+    return usage;
+}
+
+// adds 'amount' vectors to the index. could be 0
+void _add_vectors(VecSimIndex *index, long long amount) {
+    VecSimIndexInfo indexInfo = VecSimIndex_Info(index);
+    int i;
+    int64_t vec[indexInfo.d];
+    for (i = 0; i < indexInfo.d; i++)
+        vec[i] = i;
+    for (i = 0; i < amount; i++)
+        VecSimIndex_AddVector(index, vec, i);
+}
+
+// deletes 'amount' vectors from the index. could be 0 or larger from the number of vectors in the
+// index
+void _delete_vectors(VecSimIndex *index, long long amount) {
+    int i;
+    for (i = 0; i < amount; i++)
+        VecSimIndex_DeleteVector(index, i);
+}
+
+// creates a generic index, supports Broute Force and HNSW
+VecSimIndex *_create_index(VecSimAlgo algo) {
+
+    VecSimParams param;
+    param.algo = algo;
+    switch (algo) {
+    case VecSimAlgo_BF:
+        param.bfParams.blockSize = 1;
+        param.bfParams.initialCapacity = 1;
+        break;
+
+    case VecSimAlgo_HNSWLIB:
+        param.hnswParams.M = 30;
+        param.hnswParams.initialCapacity = 1;
+        param.hnswParams.efConstruction = 0;
+        param.hnswParams.efRuntime = 0;
+        break;
+    }
+    param.type = VecSimType_INT64;
+    param.size = DIMENSION;
+    param.metric = VecSimMetric_L2;
+
+    return VecSimIndex_New(&param);
+}
+
+// creates an index, add n vectors, deletes m vectors and then compares redis-server memory usage
+// report to the index memory usage report uncomment the 3 lines close to the end to get the values
+// along with the comparison result.
+int VecSim_memory_create_index_add_n_delete_m_check(RedisModuleCtx *ctx, RedisModuleString **argv,
+                                                    int argc) {
+
+    long long n, m, i;
+    long long startMemory, endMemory;
+    VecSimAlgo algo = -1;
+    RedisModuleString *al;
+
+    if (argc != 4 || RedisModule_StringToLongLong(argv[2], &n) == REDISMODULE_ERR ||
+        RedisModule_StringToLongLong(argv[3], &m) == REDISMODULE_ERR) {
+        RedisModule_WrongArity(ctx);
+        return REDISMODULE_OK;
+    }
+    al = RedisModule_CreateString(ctx, "BF", 2);
+    if (!RedisModule_StringCompare(argv[1], al))
+        algo = VecSimAlgo_BF;
+    RedisModule_FreeString(ctx, al);
+    al = RedisModule_CreateString(ctx, "HNSW", 4);
+    if (!RedisModule_StringCompare(argv[1], al))
+        algo = VecSimAlgo_HNSWLIB;
+    RedisModule_FreeString(ctx, al);
+    if (algo == -1) {
+        RedisModule_WrongArity(ctx);
+        return REDISMODULE_OK;
+    }
```

Comment:
here, you can extract the cstring from `argv[1]` using `RedisModule_StringPtrLen` and just use `strcmp`, instead of creating and deleting RedisModuleStrings.

Reply:
cool, much simpler